src/test/compile-fail/keyword-true-as-identifier.rs

```diff
@@ -9,5 +9,7 @@
 // except according to those terms.
 
 fn main() {
-    let true = "foo"; //~ error: mismatched types
+    let true = "foo";
```

Comment:
let's change this test from `"foo"` to `22` -- I feel like the message about non-reference patterns isn't central to the test.